README.md

```diff
@@ -37,143 +37,160 @@ pre-commit install
-The batch sequence can be generated using [Pulser](https://github.com/pasqal-io/Pulser). See their [documentation](https://pulser.readthedocs.io/en/stable/),
-for more information on how to install the library and create your own sequence.
+### Authentication
 
-Once you have created your sequence, you should serialize it as follows:
+There are several ways to provide a correct authentication using the SDK.
 
 ```python
-# sequence should be a pulser Sequence object
-serialized_sequence = sequence.to_abstract_repr()
+from pasqal_cloud import SDK
+
+project_id = "your_project_id"  # Replace this value with your project_id on the PASQAL platform.
+username = "your_username"  # Replace this value with your username or email on the PASQAL platform.
+password = "your_password"  # Replace this value with your password on the PASQAL platform.
 ```
 
-Once you have serialized your sequence, you can send it with the SDK with the following code
+Ideally, do not write this password in a script but provide in through the command-line or as a secret environment variable.
 
-```python
-from pasqal_cloud import SDK
-
-project_id="your_project_id" # Replace this value by your project_id on the PASQAL platform.
-username="your_username" # Replace this value by your username or email on the PASQAL platform.
-password="your_password" # Replace this value by your password on the PASQAL platform.
+#### Method 1: Username + Password
+If you know your credentials, you can pass them to the SDK instance on creation.
 
-# Initialize the cloud client
+```python
 sdk = SDK(username=username, password=password, project_id=project_id)
+```
 
-# When creating a job, select a number of runs and set the desired values for the variables
-# defined in the sequence
-job1 = {"runs": 20, "variables": {"omega_max": 6} }
-job2 = {"runs": 50, "variables": {"omega_max": 10.5} }
+#### Method 2: Username only
+If you only want to insert your username, but want a solution to have your password being secret,
+you can run the SDK without a password. A prompt will then ask for your password
 
-# Send the batch of jobs to the QPU and wait for the results
-batch = sdk.create_batch(serialized_sequence, [job1,job2], wait=True)
+```python
+sdk = SDK(username=username, project_id=project_id)
+```
 
-# You can also choose to run your batch on an emulator using the optional argument 'emulator'
-# For using a basic single-threaded QPU emulator that can go up to 10 qubits, you can specify the "EMU_FREE" emulator.
-from pasqal_cloud.device import EmulatorType
-batch = sdk.create_batch(serialized_sequence, [job1,job2], emulator=EmulatorType.EMU_FREE)
+#### Method 3: Use a custom token provider
+You can define a custom class to provide the token.
+For example, if you know your token, you can use that token to authenticate directly to our APIs as follows.
```

Comment:
`if you know your token, you can use that token to authenticate directly to our APIs as follows.` -> `if you have a token, you can use it to authenticate with our APIs.`